File: Planets/Thear/Objects.cs
using System;
using System.Collections.Generic;

public class Objects
{
    // Define the Furniture dictionary
    private static Dictionary<string, Dictionary<string, int>> Furniture = new Dictionary<string, Dictionary<string, int>>
    {
        {"Stool", new Dictionary<string, int>
            {
                {"Main Hall", 38},
                {"Back Room", 10},
                {"Training Room", 0},
                {"Office", 2},
                {"Storage Room", 0},
                {"Bedroom", 0}
            }
        },
        {"Table", new Dictionary<string, int>
            {
                {"Main Hall", 11},
                {"Back Room", 1},
                {"Training Room", 0},
                {"Office", 1},
                {"Storage Room", 0},
                {"Bedroom", 0}
            }
        },
        {"Storage", new Dictionary<string, int>
            {
                {"Main Hall", 5},
                {"Back Room", 4},
                {"Training Room", 0},
                {"Office", 0},
                {"Storage Room", 8},
                {"Bedroom", 7}
            }
        }
    };

    // Retrieves the quantity of furniture in a specific room
    public static int GetFurnitureQuantity(string type, string room)
    {
        if (Furniture.ContainsKey(type) && Furniture[type].ContainsKey(room))
        {
            return Furniture[type][room];
        }
        else
        {
            Console.WriteLine($"Invalid type ({type}) or room ({room}). Please provide valid arguments.");
            return 0;
        }
    }

    // Example usage
    public static void Main(string[] args)
    {
        Console.WriteLine(GetFurnitureQuantity("Stool", "Main Hall"));  // Output: 38
        Console.WriteLine(GetFurnitureQuantity("Table", "Office"));     // Output: 1
        Console.WriteLine(GetFurnitureQuantity("Storage", "Bedroom"));  // Output: 7
        Console.WriteLine(GetFurnitureQuantity("Chair", "Main Hall"));   // Output: 0 (Invalid type)
    }
}

File: Planets/Thear/Objects.js
// File: robjam1990/Psychosis/Gameplay/Planets/Thear/Objects.js

// Define the Furniture object outside the Objects function
const Furniture = {
    Stool: {
        "Main Hall": 38,
        "Back Room": 10,
        "Training Room": 0,
        "Office": 2,
        "Storage Room": 0,
        "Bedroom": 0
    },
    Table: {
        "Main Hall": 11,
        "Back Room": 1,
        "Training Room": 0,
        "Office": 1,
        "Storage Room": 0,
        "Bedroom": 0
    },
    Storage: {
        "Main Hall": 5,
        "Back Room": 4,
        "Training Room": 0,
        "Office": 0,
        "Storage Room": 8,
        "Bedroom": 7
    }
};

/**
 * Retrieves the quantity of furniture in a specific room.
 * @param {string} type - The type of furniture (e.g., "Stool", "Table", "Storage").
 * @param {string} room - The room where the furniture is located.
 * @returns {number} The quantity of furniture in the specified room, or 0 if the type or room doesn't exist.
 */
function getFurnitureQuantity(type, room) {
    if (Furniture[type] && Furniture[type][room]) {
        return Furniture[type][room];
    } else {
        console.error(`Invalid type (${type}) or room (${room}). Please provide valid arguments.`);
        return 0;
    }
}

// Example usage:
console.log(getFurnitureQuantity("Stool", "Main Hall")); // Output: 38
console.log(getFurnitureQuantity("Table", "Office")); // Output: 1
console.log(getFurnitureQuantity("Storage", "Bedroom")); // Output: 7
console.log(getFurnitureQuantity("Chair", "Main Hall")); // Output: 0 (Invalid type)
